blocks/variables.js

```diff
@@ -186,22 +186,21 @@ Blockly.Blocks['variables_get'] = {
       }
 
       if(topBlock.index != null){
-        // find the type of the child that has a type and set output to that type
+        // find the type of the connection that has a type and set output to that type
         var checkType;
-        var children = variableUses[topBlock.index].getChildren();
-        if(children.length == 0){
-          this.modifyBlock("None");
+
+        try{checkType = variableUses[topBlock.index].inputList[0]
+                                                    .connection
+                                                    .targetConnection
+                                                    .check_[0];
+        }catch(e){}
```

Comment:
right I'm just trying to eat the errors. I am expecting to fail often. 

if there isn't a targetConnection (which is the case when there is no block attached) then you cannot access .check_[0]. I suppose it might be more correct to check if targetConnection and check_ exist first